### ISSUE TITLE
Question about the RCE loss 

### ISSUE THREAD

**Issue by hongbo-sun** [P2333/RCE#3]
Thanks for your remarkable work. I would ask for your help about the RCE loss. According the code and the statement in your paper, I reproduce the RCE loss with pytorch carefully. However, the RCE loss would push the prediction distribution into uniform distribution. For example, if I have 200 classes, the prediction probability of the ground truth class is 1/200. The same value is observed on other classes. It seems the network gets the trivial solution with RCE loss. Could you please give me some advice? Thank you very much. Your work is really valuable.

**Comment by P2333** [comment]
Hi Hongbo, thank you for your interest.

Mathematically, during training, if we have 200 classes, the the prediction probability of the ground truth class should be zero, and of other 199 classes should be 1/199. During inference, we product -1 for all the logits to obtain correct predictions.

Numerically, larger number of classes will make the supervision of RCE loss less significant (i.e., 1/9 for 10 classes is much larger than 1/199 for 200 classes), thus it becomes more difficult to numerically distinguish between the ground truth class and other classes. A potential solution (I did not try this) is to impose a temperature term T>1 (e.g., T=100) onto the logit Wz+b, i.e., use T*(Wz+b) as the logit during training, and use Wz+b during inference. Intuitively, this temperature will magnify the gap between model prediction and reverse label, providing stronger supervision signals.

**Comment by hongbo-sun** [comment]
Thanks for your response. Have you ever tried the RCE loss in the scenario of large number of classes, which can evaluate its generalization ability.

**Comment by P2333** [comment]
I have tried CIFAR-100 for 100 classes, but not larger number of classes. As indicated in Theorem 2, the upper bounds scale with (L-1)^{2}, i.e., larger number of classes leads to looser bounds, so it may be non-trivial (requiring some numerical tricks) for RCE loss to scale to larger number of classes.

**Comment by hongbo-sun** [comment]
Does it still work under the 100 classes of CIFAR-100, which weight is 1/99 in the loss back propagation. The weight is small and near zero.

**Comment by P2333** [comment]
RCE can get non-trivial (not uniform) prediction on CIFAR-100, but I do not remember the specific accuracy that whether it can match the standard performance compared to CE.

**Comment by hongbo-sun** [comment]
OK. I got it. Thanks for your patient response. Thank you!
